history info: reintroduced merged history info

edynox opened the pull request:
Merging transactions is not working due to changes in d100a29.
Transactions are displayed sequentially. With current design,
we don't know which transactions were
given by range in command line and which were not.

ignatenkobrain commented on dnf/cli/output.py:
```diff
@@ -1498,33 +1498,25 @@ def historyInfoCmd(self, extcmds, pats=[]):
             old = self.history.last(complete_transactions_only=False)
             if old is not None:
                 tids.add(old.tid)
-                utids.add(old.tid)
```
#753  ?

edynox replied:
yep, duplicit, #753 didn't introduce all required changes

edynox commented:
Do we want to roll back to merging transactions in history info, or just stick with sequential approach? (in that case I can remove that comment block from code)

ignatenkobrain commented:
@edynox would be very cool if you can provide output which was in yum, what's now in dnf and what is now. So @kkaarreell can give his opinion.

edynox commented:
DNF merged: 
https://paste.fedoraproject.org/paste/XhP1tx0KoDQmzgbSbsci2V5M1UNdIGYhyRLivL9gydE=

DNF: sequential: https://paste.fedoraproject.org/paste/a4WYE~BUvAzGYPzCEYcF9l5M1UNdIGYhyRLivL9gydE=

YUM: 
https://paste.fedoraproject.org/paste/lB~VpB6gEUWR~lENNTis0l5M1UNdIGYhyRLivL9gydE=

kkaarreell commented:
I think that this is not so important but definitely having the functionality closer to yum behaviour would be better. Btw, is it possible to switch between these two modes? Just asking, AFAIK yum does only the merge output.

edynox commented:
Well, you could use `history info 5 6 7` instead of  `history info 5..7`, then the output will be sequential.

j-mracek commented:
test this please

edynox commented:
Can you see what's wrong with the `SideCI`?

j-mracek commented:
@edynox I don't see any details, bur can you rebase the pr and then we can see if it disappears

edynox commented:
@j-mracek done, looks good

j-mracek commented:
test this please

j-mracek commented:
Ok, I was unable to merge history output for ``dnf history info 40..42``. It provides same output  like ``dnf history info 40 41 42``.

j-mracek commented:
If you can reintroduce the merged output it would be nice due to keep yum-dnf compatibility. Or did I miss something?

edynox commented:
We would have to revert some changes from d100a29 but it should be possible. I will take a look at it.

j-mracek commented:
Thanks

edynox commented:
It should be working now.

j-mracek commented:
@edynox Please there is one test failing:  ERROR: Test history _convert_tids().

edynox commented:
@j-mracek  ah, sorry I had changed function args and forgot about unit tests... fixed
`TypeError: _convert_tids() takes exactly 2 arguments (1 given)` -> marked merged arg as optional

j-mracek commented:
Test this please

j-mracek commented:
Please still one unitest failing. 
```
1: ======================================================================
1: ERROR: Test history _convert_tids().
1: ----------------------------------------------------------------------
1: Traceback (most recent call last):
1:   File "/builddir/build/BUILD/dnf-2.4.1_1-2gab747c7/tests/mock.py", line 1199, in patched
1:     return func(*args, **keywargs)
1:   File "/builddir/build/BUILD/dnf-2.4.1_1-2gab747c7/tests/test_commands.py", line 76, in test_history_convert_tids
1:     self.assertEqual(cmd._convert_tids(), [1, 2, 3, 4, 5, 99, 113, 123])
1: TypeError: _convert_tids() takes exactly 2 arguments (1 given)
1: 
1: ----------------------------------------------------------------------
1: Ran 555 tests in 6.895s
1: 
1: FAILED (errors=1)
```

edynox commented:
sorry, I probably forgot to push that. Fixed, tests passed.

j-mracek commented:
Thanks

j-mracek commented:
@dnf-bot r+
